### PR TITLE
`Development`: Admin view for managing schools

### LINKED ISSUE #1931: Admin view for managing schools
### Description

Create admin view for managing schools, for adding and editing schools

### Tasks

_No response_

### Acceptance Criteria

_No response_

### PR DESCRIPTION
**Pull request by sachmii**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://confluence.aet.cit.tum.de/spaces/AP/pages/257786006/Language+Guidelines+Client).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311714/PR+Guidelines).

#### Server

- [x] **Important**: I implemented the changes with a [very good performance](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311718/Performance+Guidelines) and prevented too many (unnecessary) and too complex database calls.
- [x] I **strictly** followed the principle of **data economy** for all database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311877/Server+Guidelines).
- [x] I documented the Java code using JavaDoc style.

#### Client

- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the principle of **data economy** for all client-server REST calls.
- [x] I **strictly** followed the [client coding and design guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311716/Client+Guidelines).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311925/Client+Test+Guidelines).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

Closes #1931 

### Description

- Added a new admin School Management view mirroring the departments UX: searchable/sortable/paginated table with create, edit, and delete actions.
- Implemented a create/edit school dialog with validation and toast feedback, plus proper reload behavior after successful save.
- Enforced deletion rules in service logic so schools can only be deleted when no departments are linked; linked schools return a bad-request response.
- Added/extended admin school endpoints for paged admin search, update, and delete.
- Integrated navigation and localization for the new schools area (route, sidebar entry, and EN/DE translation keys for table, dialogs, buttons, and messages).
- Expanded test coverage across server and client, including duplicate-name conflict handling, delete restriction checks, and explicit null dialogRef edge-case tests in the schools component.

### Steps for Testing

Prerequisites:

1. Log in to TumApply as Admin
2. Go to new Schools view
3. Try to edit/create/delete schools (schools with linked departments should throw an error)

#### Code Review

- [x] Code Review 1

#### Manual Tests

- [x] Test 1

### Screenshots

Screenshot 2026-02-27 at 09 21 47
Screenshot 2026-02-18 at 15 22 22

### Test Coverage

**Warning:** Both client and server tests failed. Coverage could not be fully measured. Please check the [workflow logs](https://github.com/ls1intum/tum-apply/actions/runs/22555016381).

_Last updated: 2026-03-01 23:11:18 UTC_

### CODE REVIEW

File: src/main/webapp/i18n/de/research-group.json [hunk 1]
```diff
@@ -407,6 +407,66 @@
         "message": "Bist du sicher, dass du dieses Department löschen möchtest?"
       }
     },
+    "schools": {
```
Comment (Bofan-Zhu) [inline src/main/webapp/i18n/de/research-group.json]: Maybe to keep the translation files small, we could add a new admin translation file? @az108 what do you think?

File: src/main/webapp/app/usermanagement/research-group/research-group-schools/research-group-schools.component.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    }
+    const school = this.schools().find(s => s.schoolId === schoolId);
+    if (school == null) {
+      return;
+    }
+
+    const dialogRef = this.dialogService.open(SchoolEditDialogComponent, {
+      header: this.translate.instant(`${this.translationKey}.editDialog.title`),
+      width: '600px',
+      style: { background: 'var(--color-background-default)', width: '60rem' },
```
Comment (Cathy0123456789) [inline src/main/webapp/app/usermanagement/research-group/research-group-schools/research-group-schools.component.ts]: Do we still use px or rather rem?

File: src/main/webapp/app/usermanagement/research-group/research-group-schools/research-group-schools.component.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    } catch {
+      this.toastService.showErrorKey(`${this.translationKey}.toastMessages.deleteFailed`);
+    }
+  }
+
+  async onCreateSchool(): Promise<void> {
+    const dialogRef = this.dialogService.open(SchoolEditDialogComponent, {
+      header: this.translate.instant(`${this.translationKey}.createDialog.title`),
+      width: '600px',
+      style: { background: 'var(--color-background-default)', width: '60rem' },
```
Comment (Cathy0123456789) [inline src/main/webapp/app/usermanagement/research-group/research-group-schools/research-group-schools.component.ts]: same here

File: src/main/webapp/app/usermanagement/research-group/research-group-schools/research-group-schools.component.ts (hunk 2 above)
Comment (Cathy0123456789) [inline src/main/webapp/app/usermanagement/research-group/research-group-schools/research-group-schools.component.ts]: Do we still use px or rather rem? And also, aren't we setting width here twice?

**Review by Cathy0123456789 — COMMENTED** [review]
Tested on TS, we check for duplicate school names, but two different schools can have the same abbreviation just now, is that intended?

File: openapi/openapi.yaml [hunk 4]
```diff
@@ -2923,6 +2987,13 @@ components:
           type: array
           items: {$ref: '#/components/schemas/ResearchGroupDTO'}
         totalElements: {type: integer, format: int64}
```
Comment (sachmii) [inline openapi/openapi.yaml]:
```suggestion
        totalElements: {type: integer, format: int64}
    PageResponseDTOSchoolDTO:
      type: object
      properties:
        content:
          type: array
          items: {$ref: '#/components/schemas/SchoolDTO'}
        totalElements: {type: integer, format: int64}
```

File: src/main/webapp/i18n/en/research-group.json [hunk 5]
```diff
@@ -407,6 +407,69 @@
...
+          "detail": "School cannot be deleted if departments are linked."
+        }
+      },
+      "createDialog": {
+        "title": "Create School",
+        "nameLabel": "Name",
+        "namePlaceholder": "e.g. School of Engineering and Design",
+        "abbreviationLabel": "Abbreviation",
+        "abbreviationPlaceholder": "e.g. ED",
+        "abbreviationUniqueHint": "Abbreviations must be unique and cannot be reused by another school.",
```
Comment (Cathy0123456789) [inline src/main/webapp/i18n/en/research-group.json]:
```suggestion
        "abbreviationUniqueHint": "Abbreviations must be unique and cannot be used by another school.",
```

**Review by Cathy0123456789 — CHANGES_REQUESTED** [review]
Thank you, retested on TS, looks good, a small translation suggestion just

**Review by Cathy0123456789 — APPROVED** [review]
